metagraph/src/cli/transform_annotation.cpp

```diff
@@ -434,15 +434,24 @@ int transform_annotation(Config *config) {
     /**************** operations on columns *****************/
     /********************************************************/
 
-    if (config->intersect_columns) {
+    if (config->aggregate_columns) {
         logger->trace("Loading annotation...");
 
         uint64_t num_columns = get_num_columns(files, Config::ColumnCompressed);
         if (!num_columns) {
-            logger->warn("No input columns to intersect");
+            logger->warn("No input columns to aggregate");
             exit(1);
         }
 
+        const uint64_t min_cols
+            = std::max((uint64_t)std::ceil(num_columns * config->min_fraction),
+                       (uint64_t)config->min_count);
+        const uint64_t max_cols
+            = std::min((uint64_t)std::floor(num_columns * config->max_fraction),
+                       (uint64_t)config->max_count);
+
+        // TODO: set width to log2(max_cols + 1) but make sure atomic
```

Comment:
The atomic increment will overflow if the slots are not wide enough. The only way to avoid this is to rewrite `atomic_fetch_and_add` to use a compare-and-swap loop, which would make the operations more expensive.

Reply:
Yes, so this is not trivial, that's why I added a TODO.

I think it's worth it, we would save a lot of RAM which is nice.
We just need to check if the slowdown is not too bad.